src/lib/forms/TextField.js

```diff
@@ -20,15 +21,20 @@ export class TextField extends Component {
     };
   }
 
-  handleChange = (newValue) => {
+  handleChange = (e) => {
+    let newValue = e.target.value;
+    const { onChange } = this.props;
     this.setState({
       value: newValue,
       valueUpdated: true,
     });
+
+    if (onChange) onChange(e);
```

Comment:
Not be changes in this PR, for the future:
```suggestion
    onChange && onChange(e);
```